### PR TITLE
dash/cellFormat-datagrid

### PR DESCRIPTION
**Pull request by sebastianbochan**
Added cellFormatter to the dataGrid module.

### CODE REVIEW

File: ts/DataGrid/DataGridOptions.d.ts [hunk 1]
```diff
@@ -114,6 +114,14 @@ export interface ColumnOptions {
      */
     cellFormat?: string;
 
+    /**
+     * Extendable method for formatting each cell's in DataGrid.
+     *
+     * @return {string}
+     *         A string to be concatenated in to the common cell's text.
+     */
+    cellFormatter?: Function;
```
Comment (karolkolodziej) [inline ts/DataGrid/DataGridOptions.d.ts:123]: Please add a dedicated interface for this callback.

**Review by karolkolodziej — CHANGES_REQUESTED** [review]
Test is missing that's for sure. 
Also, one thing to consider if we should add the `headerFormatter`

File: ts/DataGrid/DataGridOptions.d.ts [hunk 2]
```diff
@@ -153,6 +153,14 @@ export interface ColumnOptions {
     show?: boolean;
 }
 
+export interface cellFormatterCallback {
```
Comment (karolkolodziej) [inline ts/DataGrid/DataGridOptions.d.ts:159]: Conventionally interfaces should start from the capital letter.

File: ts/DataGrid/DataGridOptions.d.ts [hunk 3]
```diff
@@ -153,6 +153,14 @@ export interface ColumnOptions {
...
+    (this: cellValue): string;
+}
+
+export interface cellValue {
```
Comment (karolkolodziej) [inline ts/DataGrid/DataGridOptions.d.ts:166]: same

**Review by karolkolodziej — CHANGES_REQUESTED** [review]

File: ts/DataGrid/DataGridOptions.d.ts [hunk 4]
```diff
@@ -145,6 +153,20 @@ export interface ColumnOptions {
...
+    (this: CellValue): string;
+}
+
+/**
+ * Value to convert
+ */
+export interface CellValue {
+    value: number|string
+}
+
```
Comment (bre1470) [inline ts/DataGrid/DataGridOptions.d.ts:169]: Shouldn't this interface be the same as DataTable's `CellType` types?

File: ts/DataGrid/DataGridOptions.d.ts [hunk 5]
```diff
@@ -145,6 +153,20 @@ export interface ColumnOptions {
     show?: boolean;
 }
 
+/**
+ * Returns a formatted call's string.
+ */
+export interface CellFormatterCallback {
+    (this: CellValue): string;
```
Comment (bre1470) [inline ts/DataGrid/DataGridOptions.d.ts:160]: The function scope makes it impossible to use arrow functions as callbacks. I would expect that the table value comes as the first or second argument.

Comment (sebastianbochan) [reply ts/DataGrid/DataGridOptions.d.ts:160]: We do not support arrow functions in Highcharts or Dashboards events and formatted.

Eg. https://jsfiddle.net/2g7edxvf/1/

Knowing that the formatter is a temporary solution (like we discussed during the last meeting), I would skip arrow function support.

File: test/cypress/integration/DataGrid/cell-format.cy.js [hunk 6]
```diff
@@ -0,0 +1,9 @@
+describe('Remove the dashboard.', () => {
+    before(() => {
+        cy.visit('/dashboards/cypress/datagrid-cell-formatter');
+    });
+
+    it('Dates un cells should be formatted.', () => {
```
Comment (goransle) [inline test/cypress/integration/DataGrid/cell-format.cy.js:6]: Minor typo here

**Review by goransle — APPROVED** [review]
LGTM (small typo in test description aside)

**Review by bre1470 — APPROVED** [review]

**Review by karolkolodziej — APPROVED** [review]

**Review by Vazonik — APPROVED** [review]
LGTM

**Pull request merged**